include/net/net_if.h

```diff
@@ -50,19 +50,33 @@ struct net_if_addr {
 	struct net_addr address;
 
 #if defined(CONFIG_NET_IPV6)
-	/** Used for track timers */
-	sys_snode_t node;
+	struct {
+		/** Used to track timers */
+		sys_snode_t node;
 
-	/** Address lifetime timer start time */
-	s64_t lifetime_timer_start;
+		/** Address lifetime timer start time */
+		u64_t timer_start;
 
-	/** lifetime timer timeout */
-	u32_t lifetime_timer_timeout;
+		/** Address lifetime timer timeout
+		 */
+		u32_t timer_timeout;
+	} lifetime;
 #endif
 
 #if defined(CONFIG_NET_IPV6_DAD)
 	/** Duplicate address detection (DAD) timer */
-	struct k_delayed_work dad_timer;
+	struct {
+		/** Used to track timers */
+		sys_snode_t node;
+
+		/** DAD timer start time */
+		u64_t timer_start;
+
+		/** DAD timer timeout. The timeout value is very short, so use
+		 * 16-bit value here.
+		 */
+		u16_t timer_timeout;
```

Comment:
> Can you give a direct answer why you don't want to do that now?

I did not say I do not want to do it, I am investigating this atm. Instead of these kind of non-constructive comments, I would suggest that you start to send patches too if you find issues.

Reply:
Please assume that I'm trying to be constructive, in particular my aim is to get rid of my -1 on this and another review. So, I'd like to know if I should just wait while you with @rveerama1 consider u32_t option (a good, strategic decision), or if instead I should concentrate on tactical issues with the patches as it is (and then my concern remains use of abs()).

All in all, some more feedback exchange would help - I don't want to be looking as I'm blocking these refactors by throwing too far-fetched ideas (IMHO they aren't, but I can't know how you feel about it, unless you tell me). Sorry for confusion and hasting, if it looks like that.

Reply:
Re: sending patches, I'm indeed in the sad situation when I spend too much time on reviews and barely can work on development myself. Well, I always told that we don't have enough knowledge sharing among IP stack developers, and asked to be a default reviewers just to address that, so I keep telling myself that I should eat that dogfood for yet another week ;-). (Until my boss asks me "what the heck are you working on, where's your code"? ;-) ).